components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -334,7 +335,16 @@ function Main() {
               <PageContextToggle />
             </div>
           )}
-          <ToolsButtonMenu {...conversationContext} />
+          <ToolsMenu
+            isOpen={conversationContext.isToolsMenuOpen}
+            setIsOpen={conversationContext.setIsToolsMenuOpen}
+            query={extractQuery(conversationContext.inputText, {
+              onlyAtStart: false,
+              triggerCharacter: '/',
+            })}
```

Comment:
Should `extractQuery` be memo-ized?

Reply:
I think it's doing a regex and I believe this will fire every time context changes, which is a lot when new delta text is added to generating conversation entries.
Could either change extractQuery to a hook (useExtractQuery) which memoizes itself, or wrap this with useMemo above?

Reply:
Wrapped it in a hook